### PR TITLE
feat: handle `Fetch.requestPaused` event

### PR DESCRIPTION
**Pull request by thiagowfx**
Bug: #644

### CODE REVIEW

File: src/bidiMapper/domains/network/NetworkRequest.ts [hunk 1]
```diff
@@ -196,7 +207,7 @@ export class NetworkRequest {
   #getBaseEventParams(): Network.BaseParameters {
     return {
       // TODO: implement.
```
Comment (OrKoN) [inline src/bidiMapper/domains/network/NetworkRequest.ts:209]: remove todo?

File: tests/network/test_add_intercept.py [hunk 2]
```diff
@@ -255,3 +257,87 @@ async def test_add_intercept_type_pattern_port_empty_invalid(websocket):
...
+        #     },
+        # ],
+    ],
+    ids=[
+        "string",
+        #  "pattern",
+        #  "string and pattern",
+    ])
+async def test_add_intercept_blocks(websocket, context_id, url_patterns):
+    url = "https://www.example.com/"
```
Comment (OrKoN) [inline tests/network/test_add_intercept.py:299]: do we want to test with example.com or perhaps set up a local server?

File: tests/network/test_add_intercept.py [hunk 3]
```diff
@@ -255,3 +257,87 @@ async def test_add_intercept_type_pattern_port_empty_invalid(websocket):
...
+    await send_JSON_command(
+        websocket, {
+            "method": "browsingContext.navigate",
+            "params": {
+                "url": url,
+                "context": context_id,
+            }
+        })
+
+    event_response = await wait_for_event(websocket, "cdp.Fetch.requestPaused")
```
Comment (OrKoN) [inline tests/network/test_add_intercept.py:325]: do we want to test using a bidi event? or alternatively add a todo.

File: tests/network/test_add_intercept.py (hunk 2 above)
Comment (thiagowfx) [reply tests/network/test_add_intercept.py:299]: We still do not have local server support: https://github.com/GoogleChromeLabs/chromium-bidi/issues/575

File: tests/network/test_add_intercept.py (hunk 3 above)
Comment (thiagowfx) [reply tests/network/test_add_intercept.py:325]: There's no BiDi event equivalent to `Fetch.requestPaused`.

And the other events (response started, response completed, etc) require other intercept commands/methods which aren't yet available.

I am considering the TODO to write more tests, but IMHO this test is OK as is - having at least one test that tests the CDP event should be fine. That said, we can add a TODO to stop using example.com

**Comment by thiagowfx** [comment]
New question: https://github.com/w3c/webdriver-bidi/issues/550

**Comment by thiagowfx** [comment]
There's still a lot to do (intended to follow up in other PRs), but I'd say this PR is ready for submission as is.

File: tests/network/test_add_intercept.py (hunk 2 above)
Comment (OrKoN) [reply tests/network/test_add_intercept.py:298]: should we address it next? we will need to have many more tests were the example.com might not be sufficient or reliable.

Comment (thiagowfx) [reply tests/network/test_add_intercept.py:298]: I agree, the more the time passes the more important this feature becomes. Let's try to prioritize it. There are some pre-existing tests that use example.com that would also benefit from local servers.

(Out of scope of this PR though.)

**Pull request merged**